### PR TITLE
XW-1279 | sort category items in alphabeticalList method

### PR DESCRIPTION
**Pull request by Warkot**
https://wikia-inc.atlassian.net/browse/XW-1279

@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/WikiaMobile/WikiaMobileCategoryService.class.php [hunk 1]
```diff
@@ -80,6 +80,7 @@ public function alphabeticalList() {
 			$title = $categoryPage->getTitle();
 			$category = Category::newFromTitle( $title );
 			$collections = $this->model->getCollection( $category, $format );
+			ksort($collections['items']);
```
Comment (rogatty) [inline extensions/wikia/WikiaMobile/WikiaMobileCategoryService.class.php]: Missing spaces :see_no_evil:

Comment (Warkot) [inline extensions/wikia/WikiaMobile/WikiaMobileCategoryService.class.php]: https://en.wikipedia.org/wiki/Seppuku :hocho: :astonished:

**Comment by rogatty** [comment]
LGTM

File: extensions/wikia/WikiaMobile/WikiaMobileCategoryService.class.php [hunk 2]
```diff
@@ -80,6 +80,7 @@ public function alphabeticalList() {
 			$title = $categoryPage->getTitle();
 			$category = Category::newFromTitle( $title );
 			$collections = $this->model->getCollection( $category, $format );
+			ksort( $collections['items'] );
```
Comment (vforge) [inline extensions/wikia/WikiaMobile/WikiaMobileCategoryService.class.php:83]: Is `ksort` - i18n-capable? Do we want the list to be sorted according to proper alphabet? @Warkot

Comment (Warkot) [inline extensions/wikia/WikiaMobile/WikiaMobileCategoryService.class.php:83]: Nope. If I understand correctly what you want to achieve—e.g. array with keys `['P', 'T', 'Ś']` sorted to look like this:

```
[
  'P' => […],
  'Ś' => […],
  'T' => […]
]
```

I didn't find any easy way for such sort by keys.

Comment (Warkot) [inline extensions/wikia/WikiaMobile/WikiaMobileCategoryService.class.php:83]: OK, after some digging I know the thing is more complex… https://www.mediawiki.org/wiki/Manual:$wgCategoryCollation

Comment (Warkot) [inline extensions/wikia/WikiaMobile/WikiaMobileCategoryService.class.php:83]: Fixed it. https://github.com/Wikia/app/pull/10111 :wrench:

**Pull request merged**